assessment_software/src/app/TextField-Component/TextField.component.ts

```diff
@@ -0,0 +1,10 @@
+import { Component } from "@angular/core";
+
+@Component({
+    selector:'app-textfield',
+    templateUrl:'./TextField.component.html',
+})
+
+export class TextField{
+    
```

Comment:
The current TextField component is static. if used, it will only show a text field that has a value "value" but is not able to change that value. 

In order, to make the TextField dynamic:
- Make the attributes of the input field dynamic and update the textfield.component.html

```
<input
        [id]="id"
        [value]="value"
        [placeholder]="placeholder"
        [autofocus]="autofocus"
        [required]="required"
        [ngClass]="classes"
        [style.background-color]="variant === 'Filled' ? '#EEE' : 'none'"
        [style.border]="variant === 'Outlined' ? '1px solid #000' : 'none'"
        [style.color]="color"
        [type]="type"
        (input)="handleChange($event.target.value)"
        aria-describedby="error ? 'error-desc' : null"
    />
    <div *ngIf="error" id="error-desc" style="color: red;">Error</div>
```
    
    in the TextField class:
    
```
export class TextField{
    @Input() id: string;
    @Input() value: string;
    @Input() variant: 'Standard' | 'Filled' | 'Outlined' = 'Standard';
    @Input() placeholder: string;
    @Input() autofocus: boolean = false;
    @Input() classes: string;
    @Input() color: string = 'black';
    @Input() error: boolean = false;
    @Input() required: boolean = false;
    @Input() type: string = 'text';
    @Output() onChange = new EventEmitter<string>();

    handleChange(value: string) {
        this.onChange.emit(value);
    }
}
```

By the way, I got this from chatgpt. I guessed how you should edit it, and then I checked with chatgpt and it had a better answer. 


But be reminded that this doesn't have any styling. We need to add CSS to make it look like the figma design. 
    
Before you make the changes, I recommend you ran the code and try to see the output and then make the changes as per the comment and ran the code again and try to see the difference.